lib/Messenger/Handler/AssetUpdateTasksHandler.php

```diff
@@ -51,51 +56,62 @@ public function __invoke(AssetUpdateTasksMessage $message): void
         }
 
         $this->longRunningHelper->deleteTemporaryFiles();
+        $this->lockFactory->createLock($asset->getUpdateQueueLockId())->release();
     }
 
-    private function saveAsset(Asset $asset): void
+    private function saveAsset(Asset $asset, array $saveParams = []): void
     {
         Version::disable();
         $asset->markFieldDirty('modificationDate'); // prevent modificationDate from being changed
-        $asset->save();
+        $asset->save($saveParams);
         Version::enable();
     }
 
     private function processDocument(Asset\Document $asset): void
     {
+        $save = false;
+        $saveParams = [];
+        $asset->removeCustomSetting(Asset::CUSTOM_SETTING_UPDATE_TASK_PROCESSING_FAILED);
         if ($asset->getMimeType() === 'application/pdf' && $asset->checkIfPdfContainsJS()) {
-            $asset->save(['versionNote' => 'PDF scan result']);
+            $save = true;
+            $saveParams['versionNote'] = 'PDF scan result';
         }
 
-        $savedNeeded = false;
         $pageCount = $asset->getCustomSetting('document_page_count');
         if (!$pageCount || $pageCount === 'failed') {
-            $savedNeeded |= $asset->processPageCount();
-        }
+            if (!$asset->processPageCount() || $asset->getCustomSetting('document_page_count') === 'failed') {
+                $asset->setCustomSetting(Asset::CUSTOM_SETTING_UPDATE_TASK_PROCESSING_FAILED, true);
+                $this->logger->warning(sprintf('Failed processing page count for document asset %s.', $asset->getId()));
+
+            } else {
+                $asset->getImageThumbnail(Asset\Image\Thumbnail\Config::getPreviewConfig())->generate(false);
+            }
 
-        if ($savedNeeded) {
-            $this->saveAsset($asset);
+            $save = true;
         }
 
-        if ($asset->getCustomSetting('document_page_count') !== 'failed') {
-            $asset->getImageThumbnail(Asset\Image\Thumbnail\Config::getPreviewConfig())->generate(false);
-        } elseif ($savedNeeded) {
-            $this->logger->warning(sprintf('Failed processing page count for document asset %s.', $asset->getId()));
+        if ($save) {
+            $this->saveAsset($asset, $saveParams);
         }
     }
 
     private function processVideo(Asset\Video $asset): void
     {
+        $asset->removeCustomSetting(Asset::CUSTOM_SETTING_UPDATE_TASK_PROCESSING_FAILED);
+        $failed = true;
```

Comment:
```suggestion
        $save = false;
```

What about something more similar to the $save (former $savedNeeded )in processDocument but for [line 131](https://github.com/pimcore/pimcore/pull/18679/files#diff-f9779d6bf7f7d2cfcbd48f036980ea510e017943bf93f2edbcc968d60689a020R131)?

Beside refactoring `$failed`, we would need to check if `SphericalMetaData` and `embeddedMetaDataExtracted` aren't changed by comparing the result with the stored getCustomSetting

Reply:
`$failed` : in `processDocument()` it's different, because there are multiple tasks, so having `false` as default and set `$save` to `true` makes more sense and works with less code, not? 

meta-data: true, it's not related to the bug fix, but still, it is/was a bug 😉 fixed that as well

Reply:
ahh ok, i wanted to prevent calling a `saveAsset()` at all when the CustomSetting data happens somehow to be just the same, but it's also fine like this with the `$failed` approach.
it's indeed more "complex" than document.